src/paths/distances.c

```diff
@@ -212,3 +238,70 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
 
     return 0;
 }
+
+/**
+ * \function igraph_pseudo_diameter
+ * \brief Approximation and lower bound of diameter.
+ *
+ * This algorithm finds a pseudo-peripheral vertex and returns its
+ * eccentricity. This value can be used as an approximation
+ * and lower bound of the diameter of a graph.
+ *
+ * </para><para>
+ * A pseudo-peripheral vertex is a vertex v, such that for every
+ * vertex u which is as far away from v as possible, v is also as
+ * far away from u as possible.
+ *
+ * \param graph The input graph, if it is directed, its edge directions
+ *   are ignored.
+ * \param p_diameter Pointer to a real variable, the result is stored
+ *   here.
+ * \return Error code.
+ *
+ * Time complexity: O(|V||E|)), where |V| is the number of
+ * vertices and |E| is the number of edges.
+ *
+ * \sa \ref igraph_eccentricity(), \ref igraph_diameter().
+ *
+ */
+int igraph_pseudo_diameter(const igraph_t *graph, igraph_real_t *p_diameter) {
```

Comment:
It would be useful to also return the endpoints of the pseudo-diameter, just like `igraph_diameter` does.